generator/src/test/java/com/scottlogic/deg/generator/cucumber/LessThan.feature

```diff
@@ -2,12 +2,332 @@ Feature: User can specify that a numeric value is lower than, but not equal to,
+Scenario: lessThan run against a contradicting lessThanOrEqualTo should only only generate string, temporal and null (lessThan 1 AND lessThanOrEqualTo 1)
+     Given foo is less than 1
+       And foo is less than or equal to 1
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
+
+Scenario: not lessThan run against a contradicting not lessThanOrEqualTo should only only generate string, temporal and null (lessThan 2 AND not lessThanOrEqualTo 3)
+     Given foo is less than 2
+       And foo is anything but less than or equal to 3
+     Then no data is created
+
+#granularTo
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a non contradicting granularTo should be successful (lessThan 2 AND granularTo 1)
+     Given foo is less than 2
+       And foo is granular to 1
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
+
+#unsure on what the output should be here
+Scenario: lessThan run against a non contradicting not granularTo should be successful (lessThan 1 AND not granularTo 1)
+     Given foo is less than 1
+       And foo is anything but granular to 1
+       And the generator can generate at most 5 rows
+     Then some data should be generated
+
+Scenario: not lessThan run against a non contradicting granularTo should be successful (not lessThan 4 AND granularTo 1)
+     Given foo is anything but less than 4
+       And foo is granular to 1
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo |
+       | 4   |
+       | 5   |
+       | 6   |
+       | 7   |
+       | 8   |
+
+#unsure on what the output should be here
+Scenario: not lessThan run against a non contradicting not granularTo should be successful (not lessThan 5 AND not granularTo 1)
+     Given foo is anything but less than 5
+       And foo is anything but granular to 1
+       And the generator can generate at most 5 rows
+     Then some data should be generated
+
+#after
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a non contradicting after should be successful (lessThan 1 AND after 2019-01-01T00:00:00.00)
+     Given foo is less than 1
+       And foo is after 2019-01-01T00:00:00.000
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
```

Comment:
Not sure about these expected results; less than 1 should be:
0
-1
-2
-3
-4

Reply:
Ordering again, they would be I think but would be last

Reply:
Ref #594